pkg/dataobj/internal/metadata/filemd/filemd.proto

```diff
@@ -14,13 +14,45 @@ option go_package = "github.com/grafana/loki/v3/pkg/dataobj/internal/metadata/fi
 message Metadata {
   // Sections within the data object.
   repeated SectionInfo sections = 1;
+
+  // A list of strings used to resolve type name references.
+  repeated string dictionary = 2;
+
+  // A list of types used by sections. The zero index is reserved for an
+  // invalid type.
+  repeated SectionType types = 3;
```

Comment:
The second level of indirection here bothers me a little, but:

* We don't know how many sections an dataobj may have in the future, and 
* We don't know how the definition of SectionType may evolve over time (top-level versioning?)

It feels safer to keep the second level of indirection to keep our encoded metadata as small as possible without needing to rework the layout again.